### PR TITLE
Fix random focus lost issues on CTRL+1/2 for a webview

### LINKED ISSUE #97992: Switch group command not working as expected
Issue Type: <b>Bug</b>

When having Python interactive window open and try to switch to that group (normally with Cmd + 2), the cursor will disappear instead of going to the "command line" (the space to write commands). Then when trying to go back to editor with Cmd + 1, this won't work again so I need to use the mouse to click where I want to go. This is happening since the last update. It's important to mention that this problem is exclusive for Python Interactive. When using editors, it works fine. 

VS Code version: Code 1.45.1 (5763d909d5f12fe19f215cbfdd29a91c0fa9208a, 2020-05-14T08:33:47.663Z)
OS version: Darwin x64 19.4.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz (4 x 2700)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 2, 2|
|Memory (System)|8.00GB (1.19GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (2)</summary>

Extension|Author (truncated)|Version
---|---|---
python|ms-|2020.5.78807
vim|vsc|1.14.3

</details>

### PR DESCRIPTION
**Pull request by rchiodo**
This PR fixes #97992

What I believe is happening (after adding a bunch of logging) is that the blur here:
https://github.com/microsoft/vscode/blob/4d9277a23a035cb0493a856ce4b1174d0db5512c/src/vs/workbench/contrib/webview/electron-browser/webviewElement.ts#L237

is also async when sent to a webview. 

This can cause the blur to fire randomly, sometimes after the focus trigger has fired (but also sometimes before, when it works).

This should resolve the problem we've been having with https://github.com/microsoft/vscode-jupyter/issues/1293 (or at least make it possible for us to handle focus correctly)

/cc @mjbvz

### CODE REVIEW

File: src/vs/workbench/contrib/webview/electron-browser/webviewElement.ts [hunk 1]
```diff
@@ -231,9 +231,11 @@ export class ElectronWebviewBasedWebview extends BaseWebview<WebviewTag> impleme
 			return;
 		}
 
-		// Clear the existing focus first.
+		// Clear the existing focus first if not already on the webview.
 		// This is required because the next part where we set the focus is async.
-		if (document.activeElement && document.activeElement instanceof HTMLElement) {
+		if (document.activeElement && document.activeElement instanceof HTMLElement && document.activeElement?.tagName !== 'WEBVIEW') {
```
Comment (rheh) [inline src/vs/workbench/contrib/webview/electron-browser/webviewElement.ts:236]: nit pick: null propagation on the last check if not required as documet.activeElement is guaranteed to exist, due to first check

Comment (rchiodo) [reply src/vs/workbench/contrib/webview/electron-browser/webviewElement.ts]: Good point. Will fix

**Comment by mjbvz** [comment]
Thanks for looking into this!

Can you make sure that this does not regress other cases, specifically:

- #101553
- #75209

I believe the existing code was written this way to handle those cases and I want to make sure we don't break them again

Make sure to test with both custom title bars enabled and with them disabled (use the `window.titleBarStyle` setting)

**Comment by rchiodo** [comment]
Thanks @mjbvz .

#75209 I had originally tested (well because my original implementation broke this bug) so that one I can say it still works. If I go to the settings icon while a webview has focus, the file theme picker will show up.

#101553 I think it still works? Does this gif show it working (meaning am I following the repro steps)?

undo

**Comment by mjbvz** [comment]
@rchiodo The looks correct to me. Thanks for testing!

I've scheduled this for the December/January vscode iteration. This should give us lots of time to test this in insiders too

**Review by rheh — APPROVED** [review]
LGTM

**Pull request merged**